ecmd-core/dll/ecmdDllCapi.C

```diff
@@ -5142,21 +5140,20 @@ uint32_t dllCreateSparseMaskFromLatchOpt(const ecmdChipTarget & i_target, const
     return rc;
 }
 
-bool findLatchInCache(const std::list<std::pair<std::string, std::string> > & i_filePairs, uint64_t i_latchHashKey64, const std::string & i_ringName, ecmdLatchBufferEntry & o_latchdata)
+bool findLatchInCache(const std::list<ecmdFileLocation> & i_fileLocs, uint64_t i_latchHashKey64, const std::string & i_ringName, ecmdLatchBufferEntry & o_latchdata)
 {
     bool foundit = false;
     std::list<ecmdLatchCacheEntry>::iterator searchCacheIter;
     std::map<uint64_t, ecmdLatchBufferEntry>::iterator searchLatchIter;
     ecmdLatchCacheEntry searchCache;
 
-    std::list<std::pair<std::string, std::string> >::const_iterator l_filePair = i_filePairs.begin();
-    while (l_filePair != i_filePairs.end())
+    for (auto l_fileLoc : i_fileLocs)
```

Comment:
This is going to cause problems with some of the old 32 bit compilers for architectures we support.